admin/cleanup/MergeDuplicateArtistCredits

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env perl
+
+=head1 NAME
+
+MergeDuplicateArtistCredits - Finds and merges duplicate artist credits
+
+=head1 SYNOPSIS
+
+MergeDuplicateArtistCredits [options]
+
+Options:
+
+    -l, --limit NUM         limit the number of entities updated
+    -d, --database NAME     name of database in DBDefs.pm to connect to
+    -n, --dry-run           show any changes but don't apply them
+    -h, --help              show this help
+
+=head1 COPYRIGHT AND LICENSE
+
+Copyright (C) 2020 MetaBrainz Foundation
+
+This file is part of MusicBrainz, the open internet music database,
+and is licensed under the GPL version 2, or (at your option) any
+later version: http://www.gnu.org/licenses/gpl-2.0.txt
+
+=cut
+
+use strict;
+use warnings;
+
+use FindBin;
+use lib "$FindBin::Bin/../../lib";
+
+use Getopt::Long qw( GetOptions );
+use List::MoreUtils qw( natatime );
+use Pod::Usage qw( pod2usage );
+use MusicBrainz::Server::Constants qw(
+    %ENTITIES
+    entities_with
+);
+use MusicBrainz::Server::Context;
+use MusicBrainz::Server::Log qw( log_info );
+
+my $limit = 100;
+my $database = 'MAINTENANCE';
+my $dry_run;
+my $help_flag;
+
+GetOptions(
+    'limit|l=i' => \$limit,
+    'database|d=s' => \$database,
+    'dry-run|n' => \$dry_run,
+    'help|h' => \$help_flag,
+);
+
+pod2usage() if $help_flag;
+pod2usage(
+    -exitval => 64, # EX_USAGE
+    -message => "$0: unrecognized arguments",
+) if @ARGV;
+
+log_info {
+    'Performing a dry run: changes reported below will not be committed ' .
+    'to the database or cache.'
+} if $dry_run;
+
+my $c = MusicBrainz::Server::Context->create_script_context(
+    database => $database,
+);
+
+$c->sql->begin;
+
+my $rows = $c->sql->select_single_column_array(q{
+    WITH ac_names AS (
+        SELECT ac.id,
+               array_agg(
+                row(acn.position,
+                    acn.artist,
+                    acn.name,
+                    acn.join_phrase)
+                ORDER BY acn.position ASC
+               ) AS names
+          FROM artist_credit ac
+          JOIN artist_credit_name acn
+            ON acn.artist_credit = ac.id
+         WHERE ac.edits_pending = 0
```

Comment:
Also, forgot to mention, but when I moved the query here I tweaked it slightly to avoid selecting artist credits with `edits_pending != 0`. I'd rather not have the script have to deal with merging the `edits_pending` columns, or breaking any open edits for that matter, and the script should pick them up again once the edits are applied.